libs/wingsdk/src/target-sim/react-website.inflight.ts

```diff
@@ -0,0 +1,97 @@
+import { ChildProcess, exec } from "child_process";
+import { writeFileSync } from "fs";
+import { join } from "path";
+import { promisify } from "util";
+import {
+  REACT_WEBSITE_TYPE,
+  ReactWebsiteAttributes,
+  ReactWebsiteSchema,
+} from "./schema-resources";
+import { IReactWebsiteClient } from "../cloud";
+import { TraceType } from "../std";
+import {
+  ISimulatorContext,
+  ISimulatorResourceInstance,
+} from "../testing/simulator";
+
+export class ReactWebsite
+  implements IReactWebsiteClient, ISimulatorResourceInstance
+{
+  private readonly context: ISimulatorContext;
+  private readonly startCommand: string;
+  private readonly path: string;
+  private readonly environments: Record<string, string>;
+  private readonly isDevRun: boolean;
+  private childProcess?: ChildProcess;
+
+  constructor(props: ReactWebsiteSchema["props"], context: ISimulatorContext) {
+    this.context = context;
+    this.path = props.path;
+    this.startCommand = props.startCommand;
+    this.environments = props.environments;
+    this.isDevRun = props.isDevRun;
+  }
+
+  public async init(): Promise<ReactWebsiteAttributes> {
+    this.addTrace(`Executing start command: ${this.startCommand}`);
+
+    writeFileSync(
+      join(`${this.path}`, "/public/wing.js"),
```

Comment:
oh that's interesting, did not thing about how to build that for aws 🤔 

 Maybe it can be solved with dotenv vault feature?

Reply:
I'm not sure... Are the env vars resolved before calling the build command? 
I know that react build is switching the env variables with strings upon the build, and then there isn't a very good way to change them (other than bypassing it using the `window` global variable or another variable, or making sure they'll be available before the build)

Reply:
Hmm... 

Alright so I misunderstood the issue. So we have 3 phases: 
- Build
- preflight
- inflight

We are building the website in `build` step, but we need data from `preflight` for building. 
That is obviously not possible, so we therefore needs to move the website building step into `preflight` instead. 

This means for `tf-aws` target, we need to build the website in Terraform. Maybe we can do it by "just" running the npm command in terraform via `external`? 
https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/external

Reply:
As far as I understood, it will bind the command to the terraform process, running it before `apply`.  At this point in time we still don't have the actual URL, (until the build is over)... so I am afraid it will have the same result (unresolved tf token bound as the value of the environment variable to the build output)

Reply:
Hmm I see. 
My terraform is not good enough for knowing the solution here (if one exist)